src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -443,47 +422,166 @@
+        }
+
+        listItemElement.className = className;
+
+        if (currentListItem.render) {
+          currentListItem.render(listItemElement, data, currentListItem)
+        } else {
+          listItemElement.appendChild(document.createTextNode(currentListItem.displayText || getText(currentListItem)));
+        }
+
+        listItemElement.hintId = i;
+      });
+
+      this.setupListScrollingPadding();
+    },
+
+    clearItemsList() {
+      this.selectedHintIndex = this.defaultSelectedHintIndex;
+      this.itemsOptions = [];
+
+      while (this.listContainer.firstChild) {
+        this.listContainer.removeChild(this.listContainer.firstChild);
+      }
+    },
+
+    setupListScrollingPadding() {
+      if (this.listContainer.scrollHeight > this.listContainer.clientHeight + 1) {
+        for (let node = this.listContainer.firstChild; node; node = node.nextSibling) {
+          node.style.paddingRight = this.cm.display.nativeBarWidth + 'px';
+        }
+      }
+    },
+
+    getListNode(index) {
+      return this.listContainer.childNodes[index];
+    },
+
+    changeActive(i, avoidWrap) {
+      if (i >= this.itemsOptions.length) {
+        i = avoidWrap ? this.itemsOptions.length - 1 : 0;
       } else if (i < 0) {
-        i = avoidWrap ? 0 : itemsList.length - 1;
+        i = avoidWrap ? 0 : this.itemsOptions.length - 1;
       }
 
-      if (this.selectedHint == i) return;
+      if (this.selectedHintIndex == i) return;
 
-      let node = this.hints.childNodes[this.selectedHint];
+      let node = this.listContainer.childNodes[this.selectedHintIndex];
 
       if (node) {
         node.className = node.className.replace(' ' + ACTIVE_HINT_ELEMENT_CLASS, '');
       }
 
-      node = this.hints.childNodes[this.selectedHint = i];
+      node = this.listContainer.childNodes[this.selectedHintIndex = i];
       node.className += ' ' + ACTIVE_HINT_ELEMENT_CLASS;
 
-      if (node.offsetTop < this.hints.scrollTop) {
-        this.hints.scrollTop = node.offsetTop - 3;
-      } else if (node.offsetTop + node.offsetHeight > this.hints.scrollTop + this.hints.clientHeight) {
-        this.hints.scrollTop = node.offsetTop + node.offsetHeight - this.hints.clientHeight + 3;
+      if (node.offsetTop < this.listContainer.scrollTop) {
+        this.listContainer.scrollTop = node.offsetTop - 3;
+      } else if (node.offsetTop + node.offsetHeight > this.listContainer.scrollTop + this.listContainer.clientHeight) {
+        this.listContainer.scrollTop = node.offsetTop + node.offsetHeight - this.listContainer.clientHeight + 3;
       }
-
-      CodeMirror.signal(this.data, 'select', itemsList[this.selectedHint], node);
     },
-
-    screenAmount() {
-      return Math.floor(this.container.clientHeight / this.hints.firstChild.offsetHeight) || 1;
-    }
   };
 
+  function getHeader(text, className = '') {
+    const header = getInitDiv(className);
+    header.innerHTML = text;
+
+    return header;
+  }
+
+  function getInitDiv(className='') {
```

Comment:
```suggestion
  function getInitDiv(className = '') {
```
Let's name it without `Div`. Moreover, let's name it with `create` instead of `get` as it really creates a new element rather than retrieving existing one.